src/main/java/website/frontrow/artificial/intelligence/PathCalculation.java

```diff
@@ -0,0 +1,120 @@
+package website.frontrow.artificial.intelligence;
+
+import java.util.ArrayList;
+
+import website.frontrow.board.Mover;
+import website.frontrow.board.Enemy;
+import website.frontrow.level.Level;
+import website.frontrow.util.Point;
+
+/**
+ * This class can calculate the best path for an AI to take 
+ * given some info. //TODO: fix comment.
+ * 
+ * This method has no use for a constructor because it always returns a path.
+ */
+@SuppressWarnings("checkstyle:hideutilityclassconstructor")
+
+public class PathCalculation 
+{
+	private static ArrayList<Integer> goal;
+	private static Point location;
+	
+	/**
+	 * Method that returns a list of all relative positions of enemies and the player.
+	 * On the X axis
+	 * @param player the player unit.
+	 * @param enemies list of all enemy units.
+	 * @param level the current level.
+	 * @return a list of all relative positions of enemies and the player, on the X axis.
+	 */
+	public static ArrayList<Integer> calculateXPath(
+			Mover player, ArrayList<Enemy> enemies, Level level)
```

Comment:
https://github.com/lkroesen/BubbleBobble/pull/43/files#diff-7e8c81e78220cf4e7e88976facc5916aR69
Here you loop over the same list of enemies again.
Rather than getting the value for every enemy, storing it in a list, returning the list, and then using that list in the next loop.

Just call `moveOnXAxis` there instead, rather than storing every value in a list.